contrib/scripts/avocado-run-inplace.sh

```diff
@@ -0,0 +1,38 @@
+#! /bin/bash -e
+#
+# shell command line helper, wrap command line ops to avocado friendly way
+#
+# Example:  avocado-run-inplace.sh dnf update -y
+#
+
+status=1
+
+while [ "$1" != "" ]; do
+	case $1 in
+		--keep)
+			keep=1
+			;;
+		*)
+			break
+	esac
+	shift
+done
+
+tmp=`mktemp -d /tmp/XXXXXXX`
+cmd="$@"
+script_name="${cmd//[^[:alnum:]-]/_}.sh"
+
+echo '#!/bin/sh' > $tmp/$script_name
+echo "pushd `pwd`" >> $tmp/$script_name
+echo "$@" >> $tmp/$script_name
+
+ln -s /bin/sh $tmp/sh
+avocado run --external-runner=$tmp/sh --external-runner-chdir=runner $script_name
+status=$?
+
+if test -z "$keep"; then
+	unlink $tmp/$script_name
+	unlink $tmp/sh
+	rmdir $tmp
+fi
+exit $status
```

Comment:
I find it amazing that you're using Avocado for such tasks and this contrib script is very welcomed. Jut to try to contest a bit the sentence "it is very script oriented", this very example could be executed with:
```
$ avocado run --external-runner /usr/bin/yum -- "update -y"
```
or with:
```
$ avocado run --external-runner "$(which yum) -y" -- update
```
or, in my system:
```
$ avocado run --external-runner "$(which sudo) dnf" -- "update -y"
```
And you can specify more "tests", which will be executed all together, like this:
```
$ avocado run --external-runner "$(which sudo) dnf install -y" -- "python2 git gcc python-devel python-pip libvirt-devel"
JOB ID     : 90a5b92e15fe59528a7d6665d7a14f3a7dff7fc1
JOB LOG    : /home/apahim/avocado/job-results/job-2017-11-22T16.26-90a5b92/job.log
 (1/1) python2 git gcc python-devel python-pip libvirt-devel: PASS (4.05 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 4.88 s
JOB HTML   : /home/apahim/avocado/job-results/job-2017-11-22T16.26-90a5b92/results.html
```
or one by one, like this:
```
$ avocado run --external-runner "$(which sudo) dnf install -y" -- python2 git gcc python-devel python-pip libvirt-devel
JOB ID     : ca06dfdabc64e11e064cc5a3edb06e323a2b28be
JOB LOG    : /home/apahim/avocado/job-results/job-2017-11-22T16.24-ca06dfd/job.log
 (1/6) python2: PASS (4.02 s)
 (2/6) git: PASS (3.73 s)
 (3/6) gcc: PASS (3.87 s)
 (4/6) python-devel: PASS (3.32 s)
 (5/6) python-pip: PASS (3.55 s)
 (6/6) libvirt-devel: PASS (3.97 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 23.70 s
JOB HTML   : /home/apahim/avocado/job-results/job-2017-11-22T16.24-ca06dfd/results.html
```

Reply:
Technically you are correct. But this is more about human habits. People type commands semi-automatically :), and resepies you purpose change structure of commands normal people use to type.  This is hard. because require thinking :)
Also, precipices you show has drawback from logging point of view. It does not show first argument (external-runner one)  in test's name. So after logs uploaded to resultsdb you'll see testcase with name 'update-y' which is not very descriptive. ./yum-update--y.sh is much better.

Reply:
Yes, you have a point :)